feat(astro): Add OAuth flow components (#3719)
clerk/javascript: .changeset/short-otters-return.md

---
"@clerk/astro": patch
---

> Add custom OAuth flow component for Astro and React

Reviewer: ```suggestion
Introduce `<AuthenticateWithRedirectCallback/>` as an Astro and as a React component
```
